lib/Finance/Calendar.pm

```diff
@@ -697,20 +685,20 @@ sub _times_dst_key {
 
 # get partial trading data for a given exchange
 sub _get_partial_trading_for {
-    my ($self, $exchange, $type) = @_;
+    my ($self, $exchange, $type, $when) = @_;
 
-    my $cached = $self->calendar->{$type};
+    my $cached          = $self->calendar->{$type};
+    my $date            = $when->truncate_to_day->epoch;
+    my $partial_defined = $cached->{$date};
 
-    my %partial_tradings;
-    foreach my $epoch (keys %$cached) {
-        foreach my $close_time (keys %{$cached->{$epoch}}) {
-            my $symbols = $cached->{$epoch}{$close_time};
-            $partial_tradings{Date::Utility->new($epoch)->days_since_epoch} = $close_time
-                if (first { $exchange->symbol eq $_ } @$symbols);
-        }
+    return unless $partial_defined;
```

Comment:
@junbon Same change in here, despite being minor this should make the code a little bit more readable.